templates/types/streaming/express/src/controllers/chat.controller.ts

```diff
@@ -59,8 +59,23 @@ export const chat = async (req: Request, res: Response) => {
         });
     };
 
-    const stream = LlamaIndexAdapter.toDataStream(response, { onFinal });
-    return streamToResponse(stream, res, {}, vercelStreamData);
+    const streamResponse = LlamaIndexAdapter.toDataStreamResponse(response, {
+      data: vercelStreamData,
+      callbacks: { onCompletion },
+    });
+    // TODO: move to LlamaIndexAdapter
+    const reader = streamResponse.body?.getReader();
+    function read() {
+      reader?.read().then(({ done, value }: { done: boolean; value?: any }) => {
+        if (done) {
+          res.end();
+          return;
+        }
+        res.write(value);
+        read();
+      });
+    }
+    read();
```

Comment:
why not using await?
```suggestion
    if (response.body) {
      const reader = response.body.getReader();
      while (true) {
        const { done, value } = await reader.read();
        if (done) {
          res.end();
          return;
        }
        res.write(value);
      }
    }
```